plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java

```diff
@@ -487,6 +493,15 @@ private S3AsyncClient createS3AsyncClient(
         return clientBuilder.build();
     }
 
+    private record S3SseContext(S3SseType sseType, Optional<String> sseKmsKeyId)
+    {
+        S3SseContext
+        {
+            requireNonNull(sseType, "sseType is null");
+            checkArgument((sseType != KMS || sseKmsKeyId.isPresent()), "sseKmsKeyId is null for SSE-KMS");
```

Comment:
unnecessary brackets.

Why is this condition differ from https://github.com/trinodb/trino/pull/22529/files#r1713664602

Reply:
This one was added firstly to validate if processing  for KMS case was correct. 
The one from config was added later, and the eagerness for config purity was leveraged.

Reply:
why not to improve this condition too?

Reply:
np, I thought that, for the sake of later processing, this basic check is just enough - you validate only case you are interested in in this point of the code, 
 thorough validation for all cases is done at the config level validation
nevertheless, I reimplemented it